client/my-sites/domains/domain-management/dataviews/dataviews-fields/domain-field.tsx

```diff
@@ -0,0 +1,83 @@
+import { PartialDomainData } from '@automattic/data-stores';
+import { PrimaryDomainLabel } from '@automattic/domains-table';
+import { domainInfoContext } from '@automattic/domains-table/src/utils/constants';
+import { getDomainTypeText } from '@automattic/domains-table/src/utils/get-domain-type-text';
+import { domainManagementLink as getDomainManagementLink } from '@automattic/domains-table/src/utils/paths';
+import { Button } from '@wordpress/components';
+import { useI18n } from '@wordpress/react-i18n';
+import { navigate } from 'calypso/lib/navigate';
+import { addQueryArgs } from 'calypso/lib/url';
+import { useDomainsDataViewsContext } from '../use-context';
+import useQueryParams from '../use-query-params';
+
+interface Props {
+	domain: PartialDomainData;
+	isAllSitesView: boolean;
+	selectedFeature?: string;
+	openDomainPane?: ( domain: PartialDomainData ) => void;
+}
+
+const DomainField = ( {
+	domain: partialDomain,
+	isAllSitesView,
+	selectedFeature,
+	openDomainPane,
+}: Props ) => {
+	const { __ } = useI18n();
+	const queryParams = useQueryParams();
+
+	const { getFullDomain, getSiteSlug } = useDomainsDataViewsContext();
+
+	const domain = getFullDomain( partialDomain );
+	const siteSlug = getSiteSlug( partialDomain );
+
+	const domainManagementLink = ! partialDomain.wpcom_domain
+		? addQueryArgs(
+				queryParams,
+				getDomainManagementLink( partialDomain, siteSlug, true, selectedFeature )
+		  )
+		: '';
+
+	const domainTypeText = domain
+		? getDomainTypeText( domain, __, domainInfoContext.DOMAIN_ROW )
+		: '';
+	const showPrimaryDomainLabel = ! isAllSitesView && domain && domain.isPrimary;
+
+	const onDomainClick = ( event: React.MouseEvent ) => {
+		event.preventDefault();
+
+		if ( ! domainManagementLink ) {
+			return;
+		}
+
+		let openInNewTab = false;
+
+		if ( event.ctrlKey || event.metaKey ) {
+			openInNewTab = true;
+		}
+
+		// Support middle click to open in new tab
+		if ( event.button === 1 ) {
+			openInNewTab = true;
+		}
+
+		if ( openDomainPane ) {
+			openDomainPane( partialDomain );
+		} else {
+			navigate( domainManagementLink, openInNewTab );
+		}
+	};
+
+	return (
+		<Button className="domains-dataviews__domain-name-button" onClick={ onDomainClick }>
+			{ showPrimaryDomainLabel && <PrimaryDomainLabel /> }
```

Comment:
When this label is rendered the field becomes too crowded. Shared a screenshot privately: p1737710599858909/1737614237.091909-slack-C082KU0C0MR

Reply:
I think you forced showing this label as it is not supposed to be shown in this context :)
But, it just copies the appearance of the cell from DomainsTable. If you force showing it there, here is what you see:
![CleanShot 2025-01-25 at 00 18 03@2x](https://github.com/user-attachments/assets/d950187a-fd31-42a0-88d1-c18fb13f2fe6)

We had a brief discussion about if we needed to bring stuff like that as it has only one usage currently, and decided to do if there is no cost for it.